SimG4Core/Application/src/RunManagerMTWorker.cc

```diff
@@ -73,7 +73,7 @@ namespace {
   int getThreadIndex() { return s_thread_index; }
 
   void createWatchers(const edm::ParameterSet& iP,
-                      SimActivityRegistry& iReg,
+                      std::unique_ptr<SimActivityRegistry>& iReg,
```

Comment:
In my opinion there should be no reason to pass a `unique_ptr` by reference, because  `unique_ptr` is all about ownership model of which this function should not have to be aware. I suggest to keep the parameter type as it is (`SimActivityRegistry&`), and change the calling code to de-reference the `std::unique_ptr<SimActivityRegistry> registry`.

Reply:
@makortel , OK